src/module.zig

```diff
@@ -176,28 +187,94 @@ pub fn Objects(comptime T: type) type {
             try dead.resize(allocator, data.capacity, true);
             try generation.ensureUnusedCapacity(allocator, 1);
 
+            if (objs.internal.updated) |*updated_fields| {
+                try updated_fields.ensureUnusedCapacity(allocator, 1);
+                updated_fields.appendAssumeCapacity(try std.bit_set.DynamicBitSetUnmanaged.initEmpty(allocator, @typeInfo(T).@"struct".fields.len));
```

Comment:
So, right now we have a single `std.bit_set.DynamicBitSetUnmanaged` per object. An interesting idea might be to use a single one for all objects.

The way this would work is we make the bitset `num_objects * @typeInfo(T).@"struct".fields.len`, then when we set bits we do `objs.internal.updated.set((object_index * @typeInfo(T).@"struct".fields.len) + field_index)`

This might be better in terms of memory efficiency, because:

```
//! DynamicBitSet:
//!   A bit set with runtime-known size, backed by an allocated slice
//!   of usize.
```

versus with the array-of-bitsets we store the `DynamicBitSetUnmanaged` (two `usize` values per bitset, plus whatever is needed for storage.)

More importantly, it would reduce the number of small allocations needed

Reply:
That said - I won't block this PR on this :).